mm2src/mm2_main/src/lp_ordermatch.rs

```diff
@@ -353,8 +358,10 @@ async fn request_and_fill_orderbook(ctx: &MmArc, base: &str, rel: &str) -> Resul
                 continue;
             },
         };
-        if let Some(keypair) = keypair {
-            if pubkey_bytes.as_slice() == keypair.public().as_ref() {
+
+        // TODO consider using `is_my_order` instead to cover ZCoin orders.
+        if let Some(ref pubkey) = my_pubkey {
+            if pubkey_bytes.as_slice() == pubkey.as_ref() {
                 continue;
             }
         }
```

Comment:
Yeah, if you want to be sure that it is your order with `ZHTLC` coin, you have to use `is_my_order`.
Now your code just repeats the first half of the `is_my_order` (`is_my_order` just use hex version of order pubkey), which was using for previous version of function:
```rust
my_pub.as_deref() == Some(order_pubkey)
```
and it didnt cover Zcoin orders.

I see, we already have orderbook [here](https://github.com/KomodoPlatform/atomicDEX-API/blob/dev/mm2src/mm2_main/src/lp_ordermatch.rs#L344), so you can use `my_p2p_pubkeys` from it for `is_my_order` func.

Reply:
Ok will do, thanks

Reply:
Everything is fine!
You put different pubkeys in right places in method:
`HashSet` first,  `internal_pubkey` second(which was `secp256k1_pubkey_hex`), and `pubkey of ask/bid` the last.

So method will compare `internal_pubkey` with `pubkey of ask/bid` or will try to find `ask/bid pubkey` in `HashSet`.
It will cover general orders and Zcoin orders.

And I see you changed `secp256k1_pubkey` (right now :D)and left documentation comments. Thanks, looks better.

I have one question. Why are you checking is it your order or not and use just `continue` if its `true`? How it affects further code?

Reply:
Thanks! :)
> Also I see you changed secp256k1_pubkey (right now :D)and left documentation comments. Thanks, looks better.

I'm not 100 percent sure, it was already written, I just noticed that it might be changed to use `is_my_order`.
In my opinion, since the public key is mine, we don't need to handle it with `process_pubkey_full_trie`, because we already have these orders in our `order trie`. I hope I understand it correctly...
> Why are you checking is it your order or not and use just continue if its true? How it affects further code?

Reply:
Oh, I checked PR, when this checking [pubkey_bytes.as_slice() == my_pubkey.as_ref()](https://github.com/KomodoPlatform/atomicDEX-API/pull/1179/files#diff-9ef9503f63124c458505e79e9d6a47eac8fb395b659dae5795bb2f77c0ad9979R298-R301) were added the first time. It was related with [zombie orders issue](https://github.com/KomodoPlatform/atomicDEX-API/issues/1148). So lets live this `is_my_order` checking, as it covers ZHTLC orders now.